Add 64bit builds

CalumMcCall opened the pull request:
This PR adds `x86_64` and `arm64-x8a` as build targets, because 64-bit builds are going to be required on Android after August 1st.

`libsodium` version was also bumped and this has been tested and works.

Commit:
Add generation of x86_64/arm64-v8a builds

Commit:
add x86_64 targets

Commit:
Fix linking and path

Commit:
update readme with x86_64

Commit:
refactor: use default installed toolchains

 - This commit changes the build system for android platforms to use the
   NDK default toolchain, instead of requiring standalone toolchains to
   be generated using the old `make_standalone_toolchain.py` script
 - This simplifies the Makefile and removes the number of env vars
   needed

Commit:
Update readme, simplify further

Commit:
fix clean and ignore superfluous warning

Commit:
Fix 64-bit compiler versions

Commit:
version bump

johnxnguyen reviewed (changes requested):
Nice work! Do you also need to modify the Makefile in the root directory? These changes are specific to Android, but perhaps for completion we should also add 64bit builds for general java distribution.

CalumMcCall commented:
I don't think I need to modify the root Makefile, as it looks like Android is built as a separate step. The JVM platform abstracts over 64-bit/32-bit for us already, so no work required to build any JVM languages differently. Can you try testing it locally and making sure things work on Mac?

Commit:
Ignore build output directories

Commit:
Use NDK toolchain to compile x86_64

marcoconti83 committed:
Add docker build scripts

marcoconti83 committed:
Disable autobuilding from Docker

This is needed temporarely, to debug

Commit:
Use rust x86_64-linux-android toolchain

Commit:
Fix cryptobox-c version

marcoconti83 committed:
Dockerfile tweaks

Commit:
add pkg-config for Rust build

Commit:
Remove unused Android SDK versions

Commit:
remove unused paths

marcoconti83 committed:
Re-add android path

marcoconti83 committed:
Re-add android home path

Commit:
Remove testing code

marcoconti83 committed:
Run docker unrestricted so I can run strace

marcoconti83 committed:
Try to compile in docker build

Commit:
Make builds less coupled

Commit:
Add PKG_CONFIG_PATH to ENV

 - This helps the rust compiler locate our platform-specific libsodium
lib builds

Commit:
Add glibc devel package required for building

marcoconti83 committed:
Fix android home variable

Commit:
Use deps with latest sodiumoxide

Commit:
bump libsodium version again

Commit:
Remove obsolete support lib

arianvp committed:
Get the armv7 build of libsodium-sys to work

We removed the other targets for now; we can parameterize later.
Some stuff still broken

arianvp committed:
Sort of fix makefile

We now get an error in Rust for x86-64

arianvp committed:
Make make make things

Only stuck on building the jar now. javac not found

marcoconti83 committed:
Fix docker build

marcoconti83 committed:
Use proper cryptobox version

marcoconti83 committed:
Update build process

Also updated the README with build instructions

marcoconti83 committed:
Add instructions to publish

marcoconti83 committed:
Merge pull request #11 from wireapp/refactor/move-to-universal-toolchain

Refactor/move to universal toolchain
